src/Python/logger.py

```diff
@@ -1,37 +1,68 @@
 # Read and Store MPU6050 Data to Disk
-
+import csv
 import serial
 from serial.serialutil import SerialException
 
 import time
 import argparse
+import os
+
+# Create timestamp to log entries and create log file
+date_str = time.strftime("%Y-%m-%d_%H-%M-%S", time.localtime())
 
 # Parse command line arguments
 parser = argparse.ArgumentParser()
 parser.add_argument("--port", type=str, default="/dev/ttyACM0")
 parser.add_argument("--baud_rate", type=int, default=115200)
+parser.add_argument("--log_dir", type=str, default="./logs")
+parser.add_argument("--csv_file_name", type=str, default=f"imu_data_{date_str}.csv")
 args = parser.parse_args()
 
+# Create log directory if it does not exist
+if not os.path.exists(args.log_dir):
+    os.makedirs(args.log_dir)
+
+# Full path to the CSV file
+file_path = os.path.join(args.log_dir, args.csv_file_name)
+
 # Open serial port
 try:
     ser = serial.Serial(args.port, args.baud_rate)
 except SerialException as ex:
     print(ex.strerror)
     exit()
 
-# Allow for Arduino to reset
+# Allow for Arduino to reset (physically press the reset button)
```

Comment:
Not sure about Mac, but in my case the Arduino resets automatically when connected to over serial.

Reply:
For Mac it's needed sadly, well it's odd if I use "screen" don't need to reset, but with the script I need to